File: CS/WindowsFormsApplication7/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;

namespace WindowsFormsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            for (int i = 0; i < 5; i++)
                dataTable1.Rows.Add(new object[] { i, "Text" + i });
        }

        private void repositoryItemPopupContainerEdit1_QueryResultValue(object sender, DevExpress.XtraEditors.Controls.QueryResultValueEventArgs e)
        {
            e.Value = memoEdit1.EditValue;
        }

        private void repositoryItemPopupContainerEdit1_QueryPopUp(object sender, CancelEventArgs e)
        {
            memoEdit1.EditValue = (sender as PopupContainerEdit).EditValue;
        }
    }
}

File: VB/WindowsFormsApplication7/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors

Namespace WindowsFormsApplication1

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            For i As Integer = 0 To 5 - 1
                dataTable1.Rows.Add(New Object() {i, "Text" & i})
            Next
        End Sub

        Private Sub repositoryItemPopupContainerEdit1_QueryResultValue(ByVal sender As Object, ByVal e As DevExpress.XtraEditors.Controls.QueryResultValueEventArgs)
            e.Value = memoEdit1.EditValue
        End Sub

        Private Sub repositoryItemPopupContainerEdit1_QueryPopUp(ByVal sender As Object, ByVal e As CancelEventArgs)
            memoEdit1.EditValue = TryCast(sender, PopupContainerEdit).EditValue
        End Sub
    End Class
End Namespace
